test/cluster/test_raft_voters.py

```
@@ -159,3 +162,86 @@ async def test_raft_limited_voters_retain_coordinator(manager: ManagerClient):
+    logging.info("Leader paused in the post-shutdown-announce window")
+
+    mark = await leader_log.mark()
+
+    # In-window the tablet-load-stats refresher (1s) wakes the voter refresher; with the bug, update_nodes() runs with
+    # self absent and demotes a healthy node while the dead leader keeps its vote. Wait for two completed refreshes
+    # rather than a fixed timeout: refreshes are serialized by the voter lock, so the second one provably started -
+    # and read the gossiper live members - after the mark, i.e. inside the window, while the first one might have
+    # already been in flight when the window opened.
+    refresh_done = 'group0_voter_handler - Voter update done'
+
+    async def two_refreshes_done():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `two_refreshes_done`

(ANN202)